packages/test/test-end-to-end-tests/src/test/orderSequentially.spec.ts

```diff
@@ -0,0 +1,391 @@
+	let sharedCell: SharedCell;
+	let sharedMap: SharedMap;
+	let changedEventData: IValueChanged[];
+	let containerRuntime: ContainerRuntime;
+	let error: Error | undefined;
+
+	const configProvider = (settings: Record<string, ConfigTypes>): IConfigProviderBase => ({
+		getRawConfig: (name: string): ConfigTypes => settings[name],
+	});
+	const errorMessage = "callback failure";
+
+	beforeEach(async () => {
+		const configWithFeatureGates = {
+			...testContainerConfig,
+			loaderProps: {
+				configProvider: configProvider({
+					"Fluid.ContainerRuntime.EnableRollback": true,
+				}),
+			},
+		};
+		container = await provider.makeTestContainer(configWithFeatureGates);
+		dataObject = await requestFluidObject<ITestFluidObject>(container, "default");
+		sharedString = await dataObject.getSharedObject<SharedString>(stringId);
+		sharedString2 = await dataObject.getSharedObject<SharedString>(string2Id);
+		sharedDir = await dataObject.getSharedObject<SharedDirectory>(dirId);
+		sharedCell = await dataObject.getSharedObject<SharedCell>(cellId);
+		sharedMap = await dataObject.getSharedObject<SharedMap>(mapId);
+
+		containerRuntime = dataObject.context.containerRuntime as ContainerRuntime;
+		changedEventData = [];
+		sharedString.on("sequenceDelta", (changed, _local, _target) => {
+			changedEventData.push(changed);
+		});
+		sharedString2.on("sequenceDelta", (changed, _local, _target) => {
+			changedEventData.push(changed);
+		});
+		sharedDir.on("valueChanged", (changed, _local, _target) => {
+			changedEventData.push(changed);
+		});
+		sharedCell.on("valueChanged", (value) => {
+			changedEventData.push(value);
+		});
+		sharedCell.on("delete", (value) => {
+			changedEventData.push(value);
+		});
+		sharedMap.on("valueChanged", (value) => {
+			changedEventData.push(value);
+		});
+	});
+
+	it("Should rollback simple edits on multiple DDS types", () => {
+		sharedString.insertText(0, "abcde");
+		sharedMap.set("key1", 0);
+		sharedCell.set(2);
+		try {
+			containerRuntime.orderSequentially(() => {
+				sharedString.removeRange(0, 5);
+				sharedMap.delete("key1");
+				sharedCell.delete();
+				throw new Error(errorMessage);
+			});
+		} catch (err) {
+			error = err as Error;
+		}
+
+		assert.notEqual(error, undefined, "No error");
+		assert.equal(error?.message, errorMessage, "Unexpected error message");
+		assert.equal(containerRuntime.disposed, false, "Container disposed");
+		assert.equal(sharedString.getText(), "abcde");
+		assert.equal(sharedMap.size, 1);
+		assert.equal(sharedMap.has("key1"), true);
+		assert.equal(sharedMap.get("key1"), 0);
+		assert.equal(sharedCell.get(), 2);
+
+		assert.equal(changedEventData.length, 9);
+		assert(
+			changedEventData[0] instanceof SequenceDeltaEvent,
+			`Unexpected event type - ${typeof changedEventData[0]}`,
+		);
+
+		assert.equal(changedEventData[1].key, "key1");
+		assert.equal(changedEventData[1].previousValue, undefined);
+
+		assert.equal(changedEventData[2], 2);
+
+		// rollback
```

Comment:
The comment is wrong since rollback is events indexed 6-8. 3-5 are the orderSequentially callback edits.